recipients_app/lib/view/pages/payments_page.dart

```diff
@@ -154,10 +157,17 @@ class _PaymentsPageState extends State<PaymentsPage> {
   }
 
   String _calculateFuturePayments(List<MappedPayment> mappedPayments) {
+    final paidOrConfirmedPayments = mappedPayments.where(
+      (payment) {
+        final paymentStatus = payment.payment.status;
+        return paymentStatus == PaymentStatus.paid || paymentStatus == PaymentStatus.confirmed;
+      },
+    ).toList();
```

Comment:
Could be extract this into a new private method named "_getAllPaidOrConfirmedPayments" and use this in both methods "_calculatePastPayments" and "_calculateFuturePayments"?

What do you think?